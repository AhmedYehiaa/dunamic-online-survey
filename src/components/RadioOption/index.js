import React from 'react';
import PropTypes from 'prop-types';


const RadioOption = ({
  name,
  value,
  label,
  handleInputChange,
}) => (
  <p className="radio-option">
    <input type="radio" id={value} name={name} onClick={() => handleInputChange(value)} />
    <label htmlFor={value}>{label}</label>
  </p>
);

RadioOption.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
};

export default RadioOption;
