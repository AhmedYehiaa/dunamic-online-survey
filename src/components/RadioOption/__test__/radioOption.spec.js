import React from 'react';
import { shallow } from 'enzyme';
import RadioOption from '../index';

it('it should renders each radio option', () => {
  const onClick = jest.fn();
  const name = 'value1';
  const selectedValue = 'value1';
  const option = {
    name: 'name1',
    value: 'value1',
  };
  const wrapper = shallow(
    <RadioOption
      key={option.value}
      name={name}
      label={option.name}
      value={option.value}
      onClick={onClick}
      defaultChecked={selectedValue === option.value}
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
