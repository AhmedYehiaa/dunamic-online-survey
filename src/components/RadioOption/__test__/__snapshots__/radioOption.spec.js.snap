// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders each radio option 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioOption
    defaultChecked={true}
    label="name1"
    name="value1"
    onClick={[MockFunction]}
    value="value1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          defaultChecked={true}
          id="value1"
          name="value1"
          onClick={[Function]}
          type="radio"
        />,
        <label
          htmlFor="value1"
        >
          name1
        </label>,
      ],
      "className": "radio-option",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "defaultChecked": true,
          "id": "value1",
          "name": "value1",
          "onClick": [Function],
          "type": "radio",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name1",
          "htmlFor": "value1",
        },
        "ref": null,
        "rendered": "name1",
        "type": "label",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            defaultChecked={true}
            id="value1"
            name="value1"
            onClick={[Function]}
            type="radio"
          />,
          <label
            htmlFor="value1"
          >
            name1
          </label>,
        ],
        "className": "radio-option",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultChecked": true,
            "id": "value1",
            "name": "value1",
            "onClick": [Function],
            "type": "radio",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name1",
            "htmlFor": "value1",
          },
          "ref": null,
          "rendered": "name1",
          "type": "label",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
